OMEdit/OMEditGUI/Editors/ModelicaEditor.cpp

```diff
@@ -390,87 +407,102 @@ void ModelicaHighlighter::initializeSettings()
-                  << "\\btype\\b"
-                  << "\\bwhen\\b"
-                  << "\\bwhile\\b"
-                  << "\\bwithin\\b";
+  QStringList keywordPatterns = getKeywords();
   foreach (const QString &pattern, keywordPatterns) {
-    rule.mPattern = QRegExp(pattern);
+    QString newPattern = QString("\\b%1\\b").arg(pattern);
+    rule.mPattern = QRegExp(newPattern);
     rule.mFormat = mKeywordFormat;
     mHighlightingRules.append(rule);
   }
   // Modelica types
-  QStringList typePatterns;
-  typePatterns << "\\bString\\b"
-               << "\\bInteger\\b"
-               << "\\bBoolean\\b"
-               << "\\bReal\\b";
+  QStringList typePatterns = getTypes();
   foreach (const QString &pattern, typePatterns) {
-    rule.mPattern = QRegExp(pattern);
+    QString newPattern = QString("\\b%1\\b").arg(pattern);
+    rule.mPattern = QRegExp(newPattern);
     rule.mFormat = mTypeFormat;
     mHighlightingRules.append(rule);
   }
 }
 
+// Function which returns list of keywords for the highlighter
+QStringList ModelicaHighlighter::getKeywords()
+{
+  QStringList keywordsList;
+  keywordsList   << "algorithm"
+                 << "and"
+                 << "annotation"
+                 << "assert"
+                 << "block"
+                 << "break"
+                 << "class"
+                 << "connect"
+                 << "connector"
+                 << "constant"
+                 << "constrainedby"
+                 << "der"
+                 << "discrete"
+                 << "each"
+                 << "else"
+                 << "elseif"
+                 << "elsewhen"
+                 << "encapsulated"
+                 << "end"
+                 << "enumeration"
+                 << "equation"
+                 << "expandable"
+                 << "extends"
+                 << "external"
+                 << "false"
+                 << "final"
+                 << "flow"
+                 << "for"
+                 << "function"
+                 << "if"
+                 << "import"
+                 << "impure"
+                 << "in"
+                 << "initial"
+                 << "inner"
+                 << "input"
+                 << "loop"
+                 << "model"
+                 << "not"
+                 << "operator"
+                 << "or"
+                 << "outer"
+                 << "output"
+                 << "optimization"
+                 << "package"
+                 << "parameter"
+                 << "partial"
+                 << "protected"
+                 << "public"
+                 << "pure"
+                 << "record"
+                 << "redeclare"
+                 << "replaceable"
+                 << "return"
+                 << "stream"
+                 << "then"
+                 << "true"
+                 << "type"
+                 << "when"
+                 << "while"
+                 << "within";
+  return keywordsList;
```

Comment:
according to the book written by peter fritzson and also from the modelica specification they are not coming under keywords.  https://www.modelica.org/documents/ModelicaSpec30.pdf (Page no.12)